packages/core/src/components/tabs2/tabs.tsx

```diff
@@ -0,0 +1,285 @@
+/*
+ * Copyright 2015 Palantir Technologies, Inc. All rights reserved.
+ * Licensed under the BSD-3 License as modified (the “License”); you may obtain a copy
+ * of the license at https://github.com/palantir/blueprint/blob/master/LICENSE
+ * and https://github.com/palantir/blueprint/blob/master/PATENTS
+ */
+
+import * as classNames from "classnames";
+import * as PureRender from "pure-render-decorator";
+import * as React from "react";
+// import { findDOMNode } from "react-dom";
+
+import { AbstractComponent } from "../../common/abstractComponent";
+import * as Classes from "../../common/classes";
+import * as Keys from "../../common/keys";
+import { IProps } from "../../common/props";
+import { safeInvoke } from "../../common/utils";
+
+import { ITab2Props, Tab2, TabId } from "./tab";
+import { generateTabPanelId, generateTabTitleId, TabTitle } from "./tabTitle";
+
+// <Tabs id="tabs">
+//     <Tab id="a" title="Alpha">
+//         <h1>first panel</h1>
+//     </Tab>
+//     <Tab id="b" title="Beta">
+//         <h1>second panel</h1>
+//     </Tab>
+//     <input type="text" placeholder="Search..." />
+// </Tabs>
+
+// TODO
+// vertical key bindings? up/dn
```

Comment:
wow you really tore this apart didn't you.

this was my feature list during development to make sure i hit everything the old `Tabs` did. if you read commit-by-commit you can watch it change 😄 . this final remaining issue is a "new" feature that i wanted some input on. after discussing with antoine, i'm gonna do it.